File: app/src/main/java/com/intelegain/agora/model/AppliedLeaveDetails.java
package com.intelegain.agora.model;

/**
 * Created by suraj.m on 1/11/17.
 */

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class AppliedLeaveDetails {

    @SerializedName("FiscalYear")
    @Expose
    public int fiscalYear;
    @SerializedName("LeaveData")
    @Expose
    public ArrayList<LeaveData> leaveData = null;


    public class LeaveData {

        @SerializedName("ID")
        @Expose
        public Integer iD;
        @SerializedName("LeaveType")
        @Expose
        public String leaveType;
        @SerializedName("LeaveFrom")
        @Expose
        public String leaveFrom;
        @SerializedName("LeaveTo")
        @Expose
        public String leaveTo;
        @SerializedName("LeaveDesc")
        @Expose
        public String leaveDesc;
        @SerializedName("LeaveStatus")
        @Expose
        public String leaveStatus;
        @SerializedName("AdminComments")
        @Expose
        public String adminComments;
        @SerializedName("TotalCL")
        @Expose
        public Integer totalCL;
        @SerializedName("TotalSL")
        @Expose
        public Integer totalSL;
        @SerializedName("TotalPL")
        @Expose
        public Integer totalPL;
        @SerializedName("TotalCO")
        @Expose
        public Integer totalCO;
        @SerializedName("ConsumedCL")
        @Expose
        public Integer consumedCL;
        @SerializedName("ConsumedSL")
        @Expose
        public Integer consumedSL;
        @SerializedName("ConsumedPL")
        @Expose
        public Integer consumedPL;
        @SerializedName("ConsumedCO")
        @Expose
        public Integer consumedCO;
        @SerializedName("BalanceCL")
        @Expose
        public Integer balanceCL;
        @SerializedName("BalanceSL")
        @Expose
        public Integer balanceSL;
        @SerializedName("BalancePL")
        @Expose
        public Integer balancePL;
        @SerializedName("BalanceCO")
        @Expose
        public Integer balanceCO;
        @SerializedName("TotalCLTillDate")
        @Expose
        public Integer totalCLTillDate;
        @SerializedName("TotalSLTillDate")
        @Expose
        public Integer totalSLTillDate;
        @SerializedName("TotalPLTillDate")
        @Expose
        public Integer totalPLTillDate;
        @SerializedName("TotalCOTillDate")
        @Expose
        public Integer totalCOTillDate;

        public int getLeaveId() {
            return this.iD;
        }

        public String getLeaveFrom() {
            return leaveFrom;
        }

        public void setLeaveFrom(String leaveFrom) {
            this.leaveFrom = leaveFrom;
        }

        public String getLeaveTo() {
            return leaveTo;
        }

        public void setLeaveTo(String leaveTo) {
            this.leaveTo = leaveTo;
        }

        public String getLeaveStatus() {
            return leaveStatus;
        }

        public String getLeaveType() {
            return leaveType;
        }

        public String getLeaveDesc() {
            return leaveDesc;
        }


    }
}

File: app/src/main/java/com/intelegain/agora/model/AppliedLeaveDetails.kt
package com.intelegain.agora.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

/**
 * Created by suraj.m on 1/11/17.
 */
class AppliedLeaveDetails {
    @kotlin.jvm.JvmField
    @SerializedName("FiscalYear")
    @Expose
    var fiscalYear = 0
    @kotlin.jvm.JvmField
    @SerializedName("LeaveData")
    @Expose
    var leaveData: ArrayList<LeaveData>? = null

    inner class LeaveData {
        @SerializedName("ID")
        @Expose
        var iD: Int? = null
        @SerializedName("LeaveType")
        @Expose
        var leaveType: String? = null
        @SerializedName("LeaveFrom")
        @Expose
        var leaveFrom: String? = null
        @SerializedName("LeaveTo")
        @Expose
        var leaveTo: String? = null
        @SerializedName("LeaveDesc")
        @Expose
        var leaveDesc: String? = null
        @SerializedName("LeaveStatus")
        @Expose
        var leaveStatus: String? = null
        @SerializedName("AdminComments")
        @Expose
        var adminComments: String? = null
        @SerializedName("TotalCL")
        @Expose
        var totalCL: Int? = null
        @SerializedName("TotalSL")
        @Expose
        var totalSL: Int? = null
        @SerializedName("TotalPL")
        @Expose
        var totalPL: Int? = null
        @SerializedName("TotalCO")
        @Expose
        var totalCO: Int? = null
        @SerializedName("ConsumedCL")
        @Expose
        var consumedCL: Int? = null
        @SerializedName("ConsumedSL")
        @Expose
        var consumedSL: Int? = null
        @SerializedName("ConsumedPL")
        @Expose
        var consumedPL: Int? = null
        @SerializedName("ConsumedCO")
        @Expose
        var consumedCO: Int? = null
        @SerializedName("BalanceCL")
        @Expose
        var balanceCL: Int? = null
        @SerializedName("BalanceSL")
        @Expose
        var balanceSL: Int? = null
        @SerializedName("BalancePL")
        @Expose
        var balancePL: Int? = null
        @SerializedName("BalanceCO")
        @Expose
        var balanceCO: Int? = null
        @SerializedName("TotalCLTillDate")
        @Expose
        var totalCLTillDate: Int? = null
        @SerializedName("TotalSLTillDate")
        @Expose
        var totalSLTillDate: Int? = null
        @SerializedName("TotalPLTillDate")
        @Expose
        var totalPLTillDate: Int? = null
        @SerializedName("TotalCOTillDate")
        @Expose
        var totalCOTillDate: Int? = null

        val leaveId: Int
            get() = iD!!

    }
}